llm/server.go

```diff
@@ -959,8 +951,8 @@ type TokenizeResponse struct {
 }
 
 func (s *llmServer) Tokenize(ctx context.Context, content string) ([]int, error) {
-	if envconfig.NewRunners() {
-		return tokenize(s.model, content)
+	if s.model != nil {
+		return llama.Tokenize(s.model, content)
```

Comment:
This isn't thread safe on all CPU architectures. Without a lock here (or an explicit memory barrier) it's possible for the cache line containing the pointer to get synced across CPUs before the data that it is pointing to even though the data was written first.